discovery/docker_discovery.go

```diff
@@ -278,38 +280,49 @@ func (d *DockerDiscovery) getContainers() {
 	d.containerCache.Prune(containerMap)
 }
 
-func (d *DockerDiscovery) manageConnection(quit chan bool) {
+func (d *DockerDiscovery) tryConnectDocker() DockerClient {
```

Comment:
This function never actually connects to Docker because until you actually make a call like `Ping()`, it doesn't actually do anything. I would rename this something like `configureDockerConnection()`.